.github/workflows/snap.yml

```diff
@@ -34,6 +34,37 @@ jobs:
         name: snap
         path: ${{ steps.snapcraft.outputs.snap }}
 
+  test-snap:
+    needs: build-snap
+    runs-on: ubuntu-latest
+    steps:
+    - id: checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - id: install-deps
+      run: |
+        sudo apt update
+        sudo apt install -y snapd gcab
+    - uses: actions/download-artifact@v4
+      id: download
+      with:
+        name: snap
+    - id: install-snap
+      run: |
+        sudo snap install --dangerous ${{ needs.build-snap.outputs.snap_name }}
+        sudo mkdir -p /var/snap/fwupd/common/var/etc/fwupd
+        sed "s,^DisabledPlugins=.*,DisabledPlugins=," /snap/fwupd/current/etc/fwupd/fwupd.conf | sudo tee /var/snap/fwupd/common/var/etc/fwupd/fwupd.conf
+    - id: build-cab
+      run: |
+        gcab --create  --nopath fakedevice124.cab data/installed-tests/fakedevice124.*
```

Comment:
`fwupdmgr firmware-build` perhaps?

Reply:
Did you mean `fwupdtool`, or actually port `firmware-build` to `fwupdmgr`?

Reply:
Bleh, fwupdtool indeed, sorry.

Reply:
I got it double wrong -- what I meant was `fwupdtool build-cabinet ARCHIVE FIRMWARE METAINFO`